src/styles/calendar.scss

```diff
@@ -0,0 +1,52 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  width: $calendar-width;
+  padding: $day-padding;
+
+  &__day {
+    box-sizing: border-box;
+    background-color: #eee;
+    border: 1px solid #000;
+    justify-content: center;
+    align-items: center;
+    width: 100px;
+    height: 100px;
+    transition-duration: 0.5s;
+    display: flex;
+
+    &::before {
+      content: '';
+
+      font-size: $day-font-size;
+      font-family: $day-font-family;
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transition-duration: 0.5s;
+      transform: translateY(-20px);
+    }
+  }
+
+  @each $day, $i in $days {
+    &--start-day-#{$day} &__day:first-child {
+      margin-left: ($day-gap + $day-size) * $i;
+    }
```

Comment:
The use of `margin-left` for positioning the first day of the month could be problematic if the `$day-gap` or `$day-size` variables are changed. Consider using a more flexible approach, such as setting the `flex` property on the `&__day` elements.